### ISSUE TITLE
`TextStyle.letterSpacing` does not work well with arabic script

### ISSUE THREAD

**Issue by Renzo-Olivares** [flutter/flutter#177406]
Skia issue: https://issues.skia.org/issues/458132646

## Issue:
`TextStyle.letterSpacing` is not handled well for languages that expect their connected letters to remain visually connected like arabic script. In the case of arabic script, space is added between connected letters making them appear visually disconnected which is not expected.

<img width="101" height="204" alt="Image" src="https://github.com/user-attachments/assets/3e16fbf2-d81a-42e6-b4c6-6836d633d096" />

### Chrome Web

Ignores letter-spacing for arabic script.

https://developer.mozilla.org/en-US/docs/Web/CSS/letter-spacing#internationalization_concerns

<img width="233" height="137" alt="Image" src="https://github.com/user-attachments/assets/f114ca71-7e81-4bd4-a459-00cbbb9e1033" />

### SwiftUI macOS

Letter-spacing is not ignored but connected letters remain visually connected.

<img width="331" height="288" alt="Image" src="https://github.com/user-attachments/assets/f76b0873-d92b-40f5-b0e8-398ce5c248ff" />

### SwiftUI iOS

Ignores letter-spacing for arabic script.

<img width="233" height="137" alt="Image" src="https://github.com/user-attachments/assets/1c27218a-e532-4499-8b0f-9a4b71b064b6" />

### Jetpack Compose Android

Ignores letter-spacing for arabic script.

<img width="243" height="356" alt="Image" src="https://github.com/user-attachments/assets/0948e16b-01c8-4849-a7ba-8611067f80d1" />
